### PR TITLE
add VPC endpoints for services used in the stack

### PR DESCRIPTION
**Pull request by chrnorm**
Adds VPC endpoints for S3 and DynamoDB.

### CODE REVIEW

File: modules/vpc/main.tf [hunk 1]
```diff
@@ -18,3 +18,123 @@ module "vpc" {
...
+  }
+}
+
+resource "aws_security_group" "rds" {
+  name_prefix = "${var.namespace}-${var.stage}-rds-vpce"
+  description = "Allow PostgreSQL inbound traffic"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    description = "TLS from VPC"
```
Comment (olkitu) [inline modules/vpc/main.tf:134]: Perhaps this description should be something like "Postgres from VPC"

**Comment by olkitu** [comment]
I think these VPC endpoints should be optional as traffic generally should be very low. Mostly no point pay VPC endpoint if not have compliance requirement to not have any internet access example. 

https://github.com/common-fate/terraform-aws-common-fate-deployment/pull/187/files#diff-eaae8e89698a5e0842f2c4c4bdffaaeaff71098a08b2c0b4ff170cc07c696da6R77-R82

https://github.com/common-fate/terraform-aws-common-fate-deployment/pull/187/files#diff-eaae8e89698a5e0842f2c4c4bdffaaeaff71098a08b2c0b4ff170cc07c696da6R54-R64

**Comment by chrnorm** [comment]
Thanks for the feedback @olkitu. I've opted to remove the additional VPC endpoints entirely for the moment to keep the module simple. Common Fate supports providing a custom VPC using the `vpc_id` variable, so if users need to customise the endpoints beyond our defaults they can provide a custom VPC to deploy Common Fate into.

**Comment by chrnorm** [comment]
Confirmed working in an internal deployment

**Pull request merged**